pkg/apis/core/validation/validation.go

```diff
@@ -5360,9 +5360,31 @@ func ValidateSecurityContext(sc *core.SecurityContext, fldPath *field.Path) fiel
 // maxGMSACredentialSpecLength is the max length, in bytes, for the actual contents
 // of a GMSA cred spec. In general, those shouldn't be more than a few hundred bytes,
 // so we want to give plenty of room here while still providing an upper bound.
+// The runAsUserName field will be used to execute the given container's entrypoint, and
+// it can be formatted as "DOMAIN/USER", where the DOMAIN is optional, maxRunAsUserNameDomainLength
+// is the max character length for the user's DOMAIN, and maxRunAsUserNameUserLength
+// is the max character length for the USER itself. Both the DOMAIN and USER have their
+// own restrictions, and more information about them can be found here:
+// https://support.microsoft.com/en-us/help/909264/naming-conventions-in-active-directory-for-computers-domains-sites-and
+//  https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.localaccounts/new-localuser?view=powershell-5.1
 const (
 	maxGMSACredentialSpecLengthInKiB = 64
 	maxGMSACredentialSpecLength      = maxGMSACredentialSpecLengthInKiB * 1024
+	maxRunAsUserNameDomainLength     = 256
+	maxRunAsUserNameUserLength       = 21
+)
+
+var (
+	// a valid NetBios Domain name cannot start with a dot, has at least 1 character,
+	// at most 15 characters, and it cannot the characters: \ / : * ? " < > |
+	validNetBiosRegex, _ = regexp.Compile(`^(?:[^\\/:\*\?"<>|\.][^\\/:\*\?"<>|]{0,14})?$`)
```

Comment:
won't this allow usernames with binary characters (including null)? how does that interact with the code paths that receive the username?

Reply:
Indeed. I was thinking that NetBios could contain most unicode characters, but apparently not, it is supposed to be ASCII characters (bytes). So, we could exclude the section \0-\32, which doesn't contain "printable" characters.  The section \128-255 is arguable, since it contains characters like: áèð etc.

Reply:
Done, added the same check for the user part as well.